solvers/c3.cc

```diff
@@ -137,23 +138,39 @@ C3::C3(const LCS& lcs, const C3::CostMatrices& costs,
   dynamics_constraints_.resize(N_);
   target_cost_.resize(N_ + 1);
   for (int i = 0; i < N_; i++) {
-    LinEq.block(0, 0, n_, n_) = lcs.A_.at(i);
-    LinEq.block(0, n_, n_, m_) = lcs.D_.at(i);
-    LinEq.block(0, n_ + m_, n_, k_) = lcs.B_.at(i);
+    LinEq.block(0, 0, n_, n_) = A_.at(i);
+    LinEq.block(0, n_, n_, m_) = D_.at(i);
+    LinEq.block(0, n_ + m_, n_, k_) = B_.at(i);
 
+    //    prog_.AddLinearEqualityConstraint(
+    //        LinEq, -d_.at(i), {x_.at(i), lambda_.at(i), u_.at(i), x_.at(i +
+    //        1)});
     dynamics_constraints_[i] =
         prog_
             .AddLinearEqualityConstraint(
-                LinEq, -lcs.d_.at(i),
+                LinEq, -d_.at(i),
                 {x_.at(i), lambda_.at(i), u_.at(i), x_.at(i + 1)})
             .evaluator()
             .get();
+    //    prog_.AddLinearConstraint(lambda_.at(i) >= VectorXd::Zero(m_));
   }
+
+  // Adding constraint for x[2] corresponding to end effector z to always be 
+  // above the ground at every time step.
+  // Impose constraint for timestep 1 and beyond to ensure that the plan doesn't go below the safe threshold/ground
+  // but allow the 0th timestep to be where the ee currently is in order to not invalidate the x_[0] == x0 constraint.
+  for (int i = 1; i < N_; i++) {
+    // WARNING!!! The x_.at(i)[2] == 0.005 constraint is hardcoded for the T example. The jack example needs line 166
+    // to be commented and line 165 to be uncommented.
+    prog_.AddLinearConstraint(x_.at(i)[2] >= options_.ee_z_state_min);
+    // prog_.AddLinearConstraint(x_.at(i)[2] == 0.005);
+  }
+
   input_costs_.resize(N_);
```

Comment:
Nothing like "WARNING!!! this is hard-coded for the T example" stuff should be in here.  This was a hack that won't take long to do the right way, so it's worth doing it for these PRs.  However it's up to you if you want to introduce this in the PR that only introduces the jack example, or if a following PR that introduces the T brings in whatever additional changes are needed in c3.cc.